18-hlt-intro.md

```diff
@@ -7,129 +7,233 @@ minutes: 45
+available in HLT2 is the same as the offline reconstruction, physics analyses
 can be done with the candidates created in HLT2. If a line is configured to be a
 Turbo line, all information on the candidates that it selects is stored in the
-raw event. These candidates can be resurrected later by the Tesla application and
-written to a microDST. This is similar to stripping streams that go to 
-microDST, where only candidates that are used in passing selections are 
-available to analysts. The Turbo stream is different because information that 
-is not saved is lost forever.
-
-We will now have a look at some of the candidates stored by the HLT. We will use the script we
-[used last time](http://lhcb.github.io/first-analysis-steps/05-interactive-dst.html)
-as a starting point, and the file
-`root://eoslhcb.cern.ch//eos/lhcb/user/r/raaij/Impactkit/00051318_00000509_1.turbo.mdst`.
-This file contains some 2016 Turbo events from [run 
-174252](http://lbrundb.cern.ch/rundb/run/174252/).  Fire up your
-favourite editor, open the script and save a copy to work on as
-`hlt_info.py`. There are a few things in the script that we don't need and can
-be removed, such as the `print_decay` method and the decay finder tools.
-
-Like the stripping, the decisions of the HLT are saved in so-called
-DecReports. You can find them in `Hlt1/DecReports` and `Hlt2/DecReports`, have a
-look at what they contain.
+raw event. The rest of the raw event, like sub-detector raw banks, is discarded,
+and cannot be recovered offline. The advantage of the Turbo stream is that
+less data are written to tape and no CPU intensive offline processing is needed.
 
-~~~ {.python}
-evt['Hlt1/DecReports']
-evt['Hlt2/DecReports']
-~~~
+The application of the software trigger is called Moore. Moore relies on the same
+algorithms as are used in Brunel to run the reconstruction and in DaVinci to
+select particle decays.
+
+### Run Moore from settings
 
-An important difference with the stripping is that for the HLT, all the
-decisions are present, even if they are false. Copy the `advance` function to
-`advance_hlt`, make it work on HLT decisions and make sure it really checks the
-decision. This can be done using the `decision` member function of a
-`DecReport`. Note that all names in the reports end with `Decision`.
+Let's start with a simple Moore script, we call it runMoore.py:
 
 ~~~ {.python}
-reports = evt['Hlt1/DecReports']
-report = reports.decReport('Hlt1TrackAllL0Decision')
-print report.decision()
+from Configurables import Moore
+# Define settings
+Moore().ThresholdSettings = "Physics_pp_2017"
+Moore().DataType = "2017"
+Moore().UseTCK = False
+Moore().RemoveInputHltRawBanks = True
+Moore().Split = ''
+# A bit more output
+Moore().EnableTimer = True
+Moore().OutputLevel = 3
+# Input data
+from PRConfig import TestFileDB
+TestFileDB.test_file_db["2016NB_25ns_L0Filt0x1609"].run(configurable=Moore())
+# Set the right L0 TCK
```

Comment:
be more explicit, e.g. `# override the TCK in the ThresholdSettings to match the input data`